openpype/tools/pyblish_pype/control.py

```diff
@@ -74,6 +74,10 @@ def _execute(self):
         item = self._items_to_process.popleft()
         item.process()
 
+        # Process events directly after the item processed. This allows to
+        # correctly show "highlighted" state for the next item to process
+        QtWidgets.QApplication.instance().processEvents()
```

Comment:
To be honest I and don't like this. It can trigger processing of next plugin directly or cause issues on different builds of Qt. If you'll change time interval back to 3 then you don't need to do it (interval 3ms == 0.003s).

Reply:
Are you sure? Isn't that exactly what it is supposed to counter?

[An interval or zero should wait for all events to process?](https://doc.qt.io/qt-5/qtimer.html#interval-prop). I'd be happy to change it back of course. Just wondering what could actually happen.

Thanks!

Reply:
> I'd be happy to change it back of course. Just wondering what could actually happen.

The whole reason of using timer with timeout is to avoid calling `processEvents`. It caused issues on other than windows platforms especially in hosts with Qt binding.

Reply:
Reverted interval back to 3 with https://github.com/pypeclub/OpenPype/pull/2646/commits/6c129320941dd316615371369b3994a8f4b1c3ef

Reply:
The goal was to remove this line with `processEvents` :)

Reply:
Unfortunately it doesn't work as intended without that line.

See:
![pyblish_pype_processEvents](https://user-images.githubusercontent.com/2439881/152505290-156c0d00-0adb-4a06-a531-bb079e2f7d35.png)

Reply:
@iLLiCiTiT It also fixes the cases where if you press the STOP button during the Extract Pointcache that takes a long time that it actually stops after it's done. Otherwise it'd still continue to Integrate Asset New.